core/controllers/ingest.py

```diff
@@ -396,6 +397,8 @@ def get_metadata(transformed_data: dict[str, pd.DataFrame]) -> dict:
 
 @transaction_retry_wrapper(max_retries=5, sleep_duration=0.6, error_type=exc.IntegrityError)
 def populate_db(
+    *,
```

Comment:
What's this wildcard doing?

Reply:
This forces everything following it to be passed in as [keyword-only arguments](https://peps.python.org/pep-3102/).

It's similar to writing `def func(*args, **kwargs)`. In that signature, all positional parameters go into the `args` argument as a tuple.

In `def func(*, **kwargs)` we're saying that we accept no positional arguments, so everything has to go into `kwargs` and be passed as keyword args.

You can also have the best of both worlds, eg `def func(x, y, *, keyword_z)` for example. Here, I think there's no particularly sensible positional args, so I'm forcing everything to be keyword args.

Reply:
Interesting!